tests/helm/helm_template_test.go

```diff
@@ -0,0 +1,132 @@
+
+	"github.com/gruntwork-io/terratest/modules/k8s"
+)
+
+func TestDevicePluginDaemonsetTemplateRenderedDeployment(t *testing.T) {
+	// Path to the helm chart we will test
+	helmChartPath, err := filepath.Abs("../../deployments/helm/nvidia-device-plugin")
+	releaseName := "nvidia-device-plugin"
+	require.NoError(t, err)
+
+	// Since we aren't deploying any resources, there is no need to setup kubectl authentication or helm home.
+
+	testCases := []struct {
+		description string
+		options     map[string]string
+		// TODO: We should find a better way to define the expected
+		expectedContainer v1.Container
+	}{
+		{
+			description: "no options",
+			expectedContainer: v1.Container{
+				SecurityContext: &v1.SecurityContext{
+					AllowPrivilegeEscalation: ptr(false),
+					Capabilities: &v1.Capabilities{
+						Drop: []v1.Capability{"ALL"},
+					},
+				},
+			},
+		},
+		{
+			description: "set compatWithCPUManager",
+			options: map[string]string{
+				"compatWithCPUManager": "true",
+			},
+			expectedContainer: v1.Container{
+				SecurityContext: &v1.SecurityContext{
+					Privileged: ptr(true),
+				},
+			},
+		},
+		{
+			description: "set mig-strategy to single",
+			options: map[string]string{
+				"migStrategy": "single",
+			},
+			expectedContainer: v1.Container{
+				SecurityContext: &v1.SecurityContext{
+					Capabilities: &v1.Capabilities{
+						Add: []v1.Capability{"SYS_ADMIN"},
+					},
+				},
+			},
+		},
+		{
+			description: "set device-list-strategy to volume-mounts",
+			options: map[string]string{
+				"deviceListStrategy": "volume-mounts",
+			},
+			expectedContainer: v1.Container{
+				SecurityContext: &v1.SecurityContext{
+					Capabilities: &v1.Capabilities{
+						Add: []v1.Capability{"SYS_ADMIN"},
+					},
+				},
+			},
+		},
+	}
+
+	for i, tc := range testCases {
+		t.Run(tc.description, func(t *testing.T) {
+			// Set up the namespace; confirm that the template renders the expected value for the namespace.
+			namespaceName := fmt.Sprintf("k8s-device-plugin-test-%d", i)
+
+			options := &helm.Options{
+				SetValues:      tc.options,
+				KubectlOptions: k8s.NewKubectlOptions("", "", namespaceName),
+				Logger:         logger.Discard,
+			}
+
+			// Run RenderTemplate to render the template and capture the output. Note that we use the version without `E`, since
+			// we want to assert that the template renders without any errors.
+			// Additionally, although we know there is only one yaml file in the template, we deliberately path a templateFiles
+			// arg to demonstrate how to select individual templates to render.
+			output := helm.RenderTemplate(t, options, helmChartPath, releaseName, []string{"templates/daemonset-device-plugin.yml"})
+
+			// Now we use kubernetes/client-go library to render the template output into the Deployment struct. This will
+			// ensure the Deployment resource is rendered correctly.
+			var deployment appsv1.Deployment
+			helm.UnmarshalK8SYaml(t, output, &deployment)
```

Comment:
Shouldn't we assert the error returned here is not `nil`?

Reply:
This function does not return an error and already asserts:
```
func UnmarshalK8SYaml(t testing.TestingT, yamlData string, destinationObj interface{}) {
	require.NoError(t, UnmarshalK8SYamlE(t, yamlData, destinationObj))
}
```

Reply:
Ah, I was incorrectly looking at `helm.UnmarshalK8sYamlE` when making this comment.